### PR TITLE
[Definition] Dont updated bundled with when 100% unnecessary

### PR DESCRIPTION
**Pull request by segiddins**
\c @indirect @andremedeiros

### CODE REVIEW

File: lib/bundler/definition.rb [hunk 1]
```diff
@@ -248,23 +248,24 @@ def lock(file, preserve_bundled_with = false)
...
-
       if @locked_bundler_version
         locked_major = @locked_bundler_version.segments.first
         current_major = Gem::Version.create(Bundler::VERSION).segments.first
 
-        if locked_major < current_major
-          Bundler.ui.warn "Warning: the lockfile is being updated to Bundler #{Bundler::VERSION.split('.').first}, " \
+        if updating_major = locked_major < current_major
+          Bundler.ui.warn "Warning: the lockfile is being updated to Bundler #{current_major}, " \
                           "after which you will be unable to return to Bundler #{@locked_bundler_version.segments.first}."
```
Comment (indirect) [inline lib/bundler/definition.rb]: Is it actually a warning that you have upgraded Bundler? I feel like the git diff will make it clear that this happened anyway...

Comment (segiddins) [inline lib/bundler/definition.rb]: this has been there since 1.10.0, and I feel like we _should_ warn when making a backwards-incompatible update

Comment (indirect) [inline lib/bundler/definition.rb]: OH that's major version. I was reading it as minor. :grimacing:

**Comment by indirect** [comment]
This makes sense to me. :ship:

File: lib/bundler/definition.rb [hunk 2]
```diff
@@ -248,23 +248,24 @@ def lock(file, preserve_bundled_with = false)
...
 
+      preserve_bundled_with ||= !updating_major && (Bundler.settings[:frozen] || !@unlocking)
+      return if lockfiles_equal?(@lockfile_contents, contents, preserve_bundled_with)
+
+      if Bundler.settings[:frozen]
+        Bundler.ui.error "Cannot write a changed lockfile while frozen."
+        return
+      end
+
       File.open(file, 'wb'){|f| f.puts(contents) }
```
Comment (andremedeiros) [inline lib/bundler/definition.rb]: Can we fix the spacing here?

**Comment by andremedeiros** [comment]
LGTM :shipit:

**Pull request merged**